crates/randcast-mock-demo/src/main.rs

```diff
@@ -0,0 +1,239 @@
+            partial_signatures,
+        );
+
+        println!("{}-res: {}", i, res);
+    });
+
+    let randomness_output = controller.get_last_output();
+
+    println!("randomness output: {}", randomness_output);
+
+    println!("finish.");
+
+    Ok(())
+}
+
+async fn run_dkg<C, S>(
+    board: &mut InMemoryBoard<C>,
+    phase0s: Vec<joint_feldman::DKG<C>>,
+    rng: &mut ThreadRng,
+) -> Vec<DKGOutput<C>>
+where
+    C: Curve,
+    // We need to bind the Curve's Point and Scalars to the Scheme
+    S: Scheme<Public = <C as Curve>::Point, Private = <C as Curve>::Scalar>,
+{
+    // Phase 1: Publishes shares
+    let mut phase1s = Vec::new();
+    for phase0 in phase0s {
+        phase1s.push(phase0.run(board, rng).await.unwrap());
+    }
+
+    // Get the shares from the board
+    let shares = board.shares.clone();
+
+    // Phase2
+    let mut phase2s = Vec::new();
+    for phase1 in phase1s {
+        phase2s.push(phase1.run(board, &shares).await.unwrap());
+    }
+
+    // Get the responses from the board
+    let responses = board.responses.clone();
+
+    let mut results = Vec::new();
+    for phase2 in phase2s {
+        results.push(phase2.run(board, &responses).await.unwrap());
+    }
+
+    // The distributed public key must be the same
+    let outputs = results
+        .into_iter()
+        .map(|res| match res {
+            Phase2Result::Output(out) => out,
+            Phase2Result::GoToPhase3(_) => unreachable!("should not get here"),
+        })
+        .collect::<Vec<_>>();
+    assert!(is_all_same(outputs.iter().map(|output| &output.public)));
+
+    outputs
+}
+
+fn setup<C, S, R: rand::RngCore>(
+    n: usize,
+    t: usize,
+    rng: &mut R,
+) -> (InMemoryBoard<C>, Vec<joint_feldman::DKG<C>>)
+where
+    C: Curve,
+    // We need to bind the Curve's Point and Scalars to the Scheme
+    S: Scheme<Public = <C as Curve>::Point, Private = <C as Curve>::Scalar>,
+{
+    // generate a keypair per participant
+    let keypairs = (0..n).map(|_| S::keypair(rng)).collect::<Vec<_>>();
+    // keypairs
+    //     .iter()
+    //     .for_each(|(private, public)| println!("{} {}", private, public));
+
+    let nodes = keypairs
+        .iter()
+        .enumerate()
+        .map(|(i, (_, public))| {
+            // println!("{}", i);
+            Node::<C>::new(i as Idx, public.clone())
+        })
+        .collect::<Vec<_>>();
+
+    // This is setup phase during which publickeys and indexes must be exchanged
+    // across participants
+    let group = Group::new(nodes, t).unwrap();
+
+    // Create the Phase 0 for each participant
+    let phase0s = keypairs
+        .iter()
+        .map(|(private, _)| joint_feldman::DKG::new(private.clone(), group.clone()).unwrap())
+        .collect::<Vec<_>>();
+
+    // Create the board
+    let board = InMemoryBoard::<C>::new();
+
+    (board, phase0s)
+}
+
+fn is_all_same<T: PartialEq>(mut arr: impl Iterator<Item = T>) -> bool {
+    let first = arr.next().unwrap();
```

Comment:
`unwrap` will panic, you can use `unwrap_or_else` add a default value, or use `ok_or_else` return an error.
refer: [Option](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else).